packages/react-components/react-dialog/e2e/Dialog.e2e.tsx

```diff
@@ -4,19 +4,224 @@ import { mount as mountBase } from '@cypress/react';
 import { FluentProvider } from '@fluentui/react-provider';
 import { teamsLightTheme } from '@fluentui/react-theme';
 
-import { Dialog } from '@fluentui/react-dialog';
+import { Dialog, DialogActions, DialogBody, DialogSurface, DialogTitle, DialogTrigger } from '@fluentui/react-dialog';
+import { Button } from '@fluentui/react-components';
 
 const mount = (element: JSX.Element) => mountBase(<FluentProvider theme={teamsLightTheme}>{element}</FluentProvider>);
 
 describe('Dialog', () => {
   it('should be closed by default', () => {
     mount(
       <Dialog>
-        <div>
-          <button id="close-btn">close</button>
-        </div>
+        <DialogTrigger>
+          <Button id="open-btn">Open dialog</Button>
+        </DialogTrigger>
+        <DialogSurface id="dialog-surface">
+          <DialogTitle>Dialog title</DialogTitle>
+          <DialogBody>
+            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam exercitationem cumque repellendus eaque
+            est dolor eius expedita nulla ullam? Tenetur reprehenderit aut voluptatum impedit voluptates in natus iure
+            cumque eaque?
+          </DialogBody>
+          <DialogActions>
+            <DialogTrigger>
+              <Button appearance="secondary">Close</Button>
+            </DialogTrigger>
+            <Button appearance="primary">Do Something</Button>
+          </DialogActions>
+        </DialogSurface>
       </Dialog>,
     );
-    cy.get('#close-btn').should('not.exist');
+    cy.get('#open-btn').should('exist');
+    cy.get('#dialog-surface').should('not.exist');
+  });
+  it('should open when trigger is clicked', () => {
+    mount(
+      <Dialog>
+        <DialogTrigger>
+          <Button id="open-btn">Open dialog</Button>
+        </DialogTrigger>
+        <DialogSurface id="dialog-surface">
+          <DialogTitle>Dialog title</DialogTitle>
+          <DialogBody>
+            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam exercitationem cumque repellendus eaque
+            est dolor eius expedita nulla ullam? Tenetur reprehenderit aut voluptatum impedit voluptates in natus iure
+            cumque eaque?
+          </DialogBody>
+          <DialogActions>
+            <DialogTrigger>
+              <Button appearance="secondary">Close</Button>
+            </DialogTrigger>
+            <Button appearance="primary">Do Something</Button>
+          </DialogActions>
+        </DialogSurface>
+      </Dialog>,
+    );
+    cy.get('#open-btn').click();
+    cy.get('#dialog-surface').should('exist');
```

Comment:
nit: might want to hoist all these selectors, they are used quite a few times

Reply:
I would also use more specific selectors to the component like `[aria-haspopup]` which represent the actual DOM markup